doc/classes/MarginContainer.xml

```diff
@@ -4,22 +4,23 @@
 		Simple margin container.
 	</brief_description>
 	<description>
-		Simple margin container. Adds a left margin to anything contained.
+		Simple margin container. Adds a top,left,bottom,right margins to anything contained.
+		[b]Note:[/b] Be careful, [Control] margin values are different than the constant margin values. See [constants] for more info.
 	</description>
 	<tutorials>
 	</tutorials>
 	<methods>
 	</methods>
 	<constants>
+		<constant name="margin_bottom" value="0" enum="margin_bottom">
```

Comment:
You shouldn't change the xml files' structure, they're generated from Godot itself. They're theme items: these values aren't constants, they're properties that can be overridden by theme resources or by the user, per-node.

The UI system can be a bit complicated to get into. For good reasons though: it's the system that powers Godot's editor, it's designed so you can create complex interfaces with it.

Reply:
Alright, from what i understand the doc builder of godot is listing `custom constants` as `theme properties`, but are we sure this is correct? I mean, i can't see any obvious way on how you could change these values by adding a theme. Also, I don't see how looking at the docs and seeing them listed as "theme properties" could help me change them. On the contrary, this seems to be misleading.

Reply:
I would've not figured out how to set these in the code if i had not checked the source code.

Reply:
So maybe the do maker should instead create a `custom_constants` category? Unless i am mistaken and these can actually be set by the theme.

Reply:
Constants can actually be edited by theme.

Reply:
> Constants can actually be edited by the theme.

@YeldhamDev cool! Can you please provide an example of how I could change the constants by using a theme in this case so that i can fix the descriptions.

Reply:
While they can be edited by the theme, I think it would still be better to use a simple `set()` rather than creating a theme just for that.

Reply:
I'v already tried creating a theme with no luck so far.

Reply:
> Alright, from what i understand the doc builder of godot is listing custom constants as theme properties, but are we sure this is correct? I mean, i can't see any obvious way on how you could change these values by adding a theme.

Of course, it's correct. These aren't constants as you can change their values.

> So maybe the doc maker should instead create a custom_constants category? Unless i am mistaken and these can actually be set by the theme.

It seems you're still learning how to use Godot. You may uncover bugs at times, but it's also likely at this point that you haven't figured out how something works yet. 

How about asking us how instead of saying that it doesn't work? It would save us all time.

Here's the theme we use at gdquest for our debug interfaces, with the MarginContainer properties set:

![Screenshot from 2019-06-25 08-31-55](https://user-images.githubusercontent.com/12694995/60058556-bd793e80-9723-11e9-95bb-bd1c1d8c9580.png)

You can use the theme editor to add editable properties to a theme. It appears in the bottom panel when you open a theme resource in the editor. Click the edit theme button and "Edit theme":

![Screenshot from 2019-06-25 08-31-13](https://user-images.githubusercontent.com/12694995/60058588-e7cafc00-9723-11e9-99bf-43fa276681a0.png)